### PR TITLE
fixed fastify spec builders, added unit tests for code as well

### LINKED ISSUE #926: Fastify GA: Fix the spec builders to properly define the keys
[NOTE]: # ( ^^ Provide a general summary of the issue in the title above. ^^ )

## Description

[NOTE]: # ( Describe the problem you're encountering. )
[TIP]:  # ( Do NOT share sensitive information, whether personal, proprietary, or otherwise! )

Fastify instrumentation has spec builders for both middleware and route handlers.  During the prototype #916, I discovered nesting of spans wasn't correct and the request wasn't properly found at times.  

Here are the prototypes of the fixes:
 * [Fix req retrieval](https://github.com/newrelic/node-newrelic/pull/916/files#diff-14052e9fd014517ec6c86ae354ceb9159a71ddca653bfec1836bda033db39d8aR8)
 * [next, and type properly defined](https://github.com/newrelic/node-newrelic/pull/916/files#diff-14052e9fd014517ec6c86ae354ceb9159a71ddca653bfec1836bda033db39d8aR100)

## Expected Behavior
All middleware should be siblings, even the route handler.  Currently they just nest.

[NOTE]: # ( Tell us what you expected to happen. )

## [Troubleshooting](https://discuss.newrelic.com/t/troubleshooting-frameworks/108787) or [NR Diag](https://docs.newrelic.com/docs/using-new-relic/cross-product-functions/troubleshooting/new-relic-diagnostics) results

[NOTE]: # ( Provide any other relevant log data. )
[TIP]:  # ( Scrub logs and diagnostic information for sensitive information )

## Steps to Reproduce
 * Run a fastify app with newrelic agent v8.x with the fastify feature flag
 * View the transaction traces

[NOTE]: # ( Please be as specific as possible. )
[TIP]:  # ( Link a sample application that demonstrates the issue. )

## Additional context
The prototype includes `segmentPrefix`. We will tackle this in #924

### PR DESCRIPTION
**Pull request by bizob2828**
## Proposed Release Notes
 * Fixed the WebFramework spec definitions for Fastify middleware and route handlers to properly retrieve the IncomingMessage from a request object.  Also, added proper definition to middleware handlers so that the relationship to consecutive middleware and route handler are siblings and not direct children.

## Links
Closes #926 

## Details
There is overlap with this PR and https://github.com/newrelic/node-newrelic/pull/931/files.  The tests will need to be reconciled.  We also will be adding more test cases scenarios when this MMF is close to being done in #928

### CODE REVIEW

File: test/versioned/fastify/fastify-2-naming.tap.js [hunk 1]
```diff
@@ -63,16 +64,24 @@ let testCount = 0
 const testUri = (uri, agent, test, port) => {
   agent.on('transactionFinished', (transaction) => {
     testCount++
-    test.equals(
+    test.equal(
```
Comment (michaelgoin) [inline test/versioned/fastify/fastify-2-naming.tap.js:67]: Do we have any tests that verify we are not going to end up with any raw route parameters or query string items in the naming?

I'm assuming this works but not sure if we have any coverage. These seem to just be hitting simple defined routes VS say '/person/111222333/' and ensuring we end up with '/person/:id' as the name (or whatever matches fastify-style).

If we don't have any coverage, perhaps should have a story for that.

File: lib/instrumentation/fastify.js [hunk 2]
```diff
@@ -47,9 +47,12 @@ const setupMiddlewareHandlers = (shim, fastify) => {
     return function wrappedFastifyUser() {
       const args = shim.argsToArray.apply(shim, arguments)
       const middlewareFunction = args[0]
```
Comment (michaelgoin) [inline lib/instrumentation/fastify.js:49]: You seem to be using destructuring everywhere else so just mentioning you could do that here too (but I don't care if you do or don't).

**Review by michaelgoin — APPROVED** [review]
👍

File: test/versioned/fastify/fastify-2-naming.tap.js (hunk 1 above)
Comment (bizob2828) [reply test/versioned/fastify/fastify-2-naming.tap.js:67]: we don't but i'll add to this https://github.com/newrelic/node-newrelic/issues/928 which was to flesh out our integration tests for fastify

**Review by michaelgoin — APPROVED** [review]

**Pull request merged**